packages/custom_lint_core/test/assist_test.dart

```diff
@@ -58,6 +64,36 @@ void main() {
     );
   });
 
+  test('CustomAssist.testRun', () async {
+    final assist1 = MyCustomAssist('CustomAssist', 'custom_1.txt');
+    final assist2 = MyCustomAssist('AnotherCustom', 'custom_2.txt');
+
+    final file = writeToTemporaryFile('''
+void main() {
+  print('Custom world');
+}
+''');
+    final result = await resolveFile2(path: file.path);
+    result as ResolvedUnitResult;
+
+    final changeList1 = await assist1.testRun(result, SourceRange.EMPTY);
+    final changeList2 = await assist2.testRun(result, SourceRange.EMPTY);
+
+    expect(
+      changeList1,
+      matcherNormalizedPrioritizedSourceChangeSnapshot('custom.json'),
+    );
+    expect(jsonEncode(changeList1[0]).contains('custom_1.txt'), true);
```

Comment:
Don't jsonEncode the object and instead read the map property corresponding to the file path

Reply:
I added **_extractFileName()** method that reads the nested maps to extract the file path value.